source/_posts/2018-01-09-automatically-managing-personal-and-work-git-configurations.markdown

```diff
@@ -0,0 +1,87 @@
+
+At work most of the repositories are private, but at home most of my work is public, which means that my work email would be public, I don't like that. So in this post I will share one way you can forget about having to switch and let the computer work for you and do it automatically.
+
+<!--more-->
+
+## Step 1: Configure your SSH keys
+
+If you are using SSH to interact with you Git server (probably GitHub) read on, otherwise skip this step. Or even better: **Start using SSH**.
+
+Let's say you have two ssh key-pairs, one for `user@work.com` and one for `user@personal.com`. You want that when pushing, cloning, etc. from a personal or a work repository you use your corresponding ssh key-pair. That's easy, modify or create a `~/.ssh/config` file to look similar to this:
+
+{% codeblock lang:bash %}
+#personal
+Host github.com-personal
+  HostName github.com
+  User personal
+  IdentityFile ~/.ssh/id_rsa_personal
+  PreferredAuthentications publickey
+
+#work
+Host github.com-work
+  HostName github.com
+  User work
+  IdentityFile ~/.ssh/id_rsa_work
+  PreferredAuthentications publickey
+{% endcodeblock %}
+
+Replace `personal` with your personal git user and `work` with your user from work and this step if done.
+
+## Step 2: Add user details to work and personal folders
+
+In this step I am assuming you have all your work stuff under a `~/work` folder and all your personal stuff under a `~/personal` folder. If you don't, then you either have to move things on your computer or adapt this step to your needs.
+
+In each folder add one file called `.gitconfig`, you should have one file `~/work/.gitconfig` and one `~/personal/.gitconfig`. In each file you should fill in your information accordingly, for example:
+
+{% codeblock lang:bash %}
+[user]
+  email = user@work.com
+  name = user
+{% endcodeblock %}
+
+{% codeblock lang:bash %}
+[user]
+  email = user@personal.com
+  name = user
+{% endcodeblock %}
+
+Second step, done! Let's move to the final step were we glue everything together.
+
+## Final step: Overriding git configuration based on the current folder
+
+In this step we are going to tell Git to use different configurations depending on where you are working on. First you need to modify or create the `~/.gitconfig` file and make it look something like this:
+
+{% codeblock lang:bash %}
+[includeIf "gitdir:~/work/"]
+    path = ~/work/.gitconfig
+
+[includeIf "gitdir:~/personal/"]
+    path = ~/personal/.gitconfig
+{% endcodeblock %}
+
+This tells git load the `~/work/.gitconfig` file when you are in any folder under `~/work` and the other configuration when you are under `~/personal`.
```

Comment:
I feel like it would be more clear if it was "This tells git **to** load"

Reply:
You could also use: `This tells git:`